### ISSUE TITLE
Ships' single points of failure

### ISSUE THREAD

**Issue by commandz0** [commandblockguy/trek-datapack#6]
Currently, it's very easy to take a ship entirely out of commission by destroying its torpedo bay or its transporter.

There are several ways to prevent this.

Firstly, redundant systems could be built. Additional torpedo bays could be built in harder-to-hit locations, to make it more difficult to completely take out the offensive capability of a ship. Staircases could be added in case an elevator is hit.

Materials could also be available on the ship for repairs. Currently, the ships have basically no survival considerations on them besides anvils. Adding a crafting bench, plus raw materials in various locations would incentivize players to explore their ships, or loot the enemy's ships, and provide a way to repair systems. Solid blocks could be used to repair holes in the floor, and other materials could be used to fix damaged systems. I don't think it would be best to provide players with buttons or levers directly, but instead, require them to jury-rig electrical wiring to repair command blocks that are missing buttons.

**Comment by acagliano** [comment]
Add blocks with the "repairShip" tag, that upon placement, deletes the block and copies all blocks within, like, a 5x5x5 or 10x10x10 radius from the master ship to the current location?

**Comment by commandz0** [comment]
It's kinda hard to detect block placement - I also feel like that would kinda remove the need to destroy the enemy's ship. In my opinion, it shouldn't be possible to make any function of the ship inoperable until a few minutes into the game, but damage should make the ship much harder to use - otherwise it would be pointless to damage it except to take down the shields.

**Comment by acagliano** [comment]
> It's kinda hard to detect block placement - I also feel like that would kinda remove the need to destroy the enemy's ship. In my opinion, it shouldn't be possible to make any function of the ship inoperable until a few minutes into the game, but damage should make the ship much harder to use - otherwise it would be pointless to damage it except to take down the shields.

Ok how about this as a fix for transporters. 
For friendly players: Allow entering the ship via the teleporter to place players on the bridge instead of the teleporter room (I may move the entrance portals to the bridge anyway). Allow the replicators to very rarely spawn "personal teleporters" which are an item that, when "used" teleports you to the transport site you would go to when using the teleporter, but you get hit with Instant Damage 1 or 2 for using it.
For unfriendly players: Upon teleporting to a place that is unsafe, select the closest safest location, OR spawn the player at the location of transport with momentary levitation or antigravity, and give them a single ender pearl.

That would address transporter failure.

Another option would be to do this:
Give to team members a "book" with various ship interfacing options:
1. Recharge shields
2. Repair sites
3. Check stats
4. ...

Basically rather than having "shield fuel" you would simply have "ship energy" and doing any action on the ship costs energy. The energy is replenished much the same way you currently fuel the shields. Actions that would cost ship energy include:
1. Firing phasers
2. Firing torpedo
3. Using the teleporters
4. Recharging shields
5. Repairing sites

Each action would have an energy cost. Clicking on [Repair Sites] would randomly choose one damaged section of the ship and respawn it, with damaged sections being: Transporter Room, Torpedo Bay, Phaser Arrays, etc. This would require you to allocate your ship energy appropriately.

**Comment by commandz0** [comment]
That seems like it would be complicated to implement. I do think that people should have to actually wander around the ship to do things, and think that parts of the ship becoming unusable is a good thing - I just don't think it should be possible to knock any one bit of functionality with a single shot. The best way to solve this is to probably go around the ship and add a few extra copies of things like replicators and extra hallways so that there's more than one way to get to any position.
I think the transporter is so critical to gameplay that the actual part where you land should be indestructible like the command blocks are, and the area around it reinforced. These can all be done with only modifications to the ship.

**Comment by acagliano** [comment]
Well i have padded them in a few layers of end stone. They should withstand a bit better
